docs/docs/api-reference/core/useRecoilSnapshotAndSubscribe.md

```diff
@@ -0,0 +1,25 @@
+---
+title: useRecoilSnapshotAndSubscribe()
+sidebar_label: useRecoilSnapshotAndSubscribe()
+---
+
+This hook synchronously returns a [`Snapshot`](/docs/api-reference/core/Snapshot) object during rendering and subscribes the calling component for all Recoil state changes.  You may want to use this hook for debug tools, or for server-side rendering where you need to synchronously have the state during the initial render.
+
+```jsx
+function useRecoilSnapshotAndSubscribe(): Snapshot
+```
+
+Be careful using this hook because it will cause the component to rerender for *all* Recoil state changes.   In the future we hope to provide the ability to debounce for performance.
```

Comment:
nit: not sure if docusaurus will format this correctly, but there are 3 spaces after the period vs the usual 2